File: src/main/java/it/unibo/crabinv/Model/PowerUpsShop/ShopLogic.java
package it.unibo.crabinv.Model.PowerUpsShop;

import it.unibo.crabinv.Model.save.UserProfile;

public class ShopLogic implements Shop {
    @Override
    public boolean purchase(UserProfile profile, PowerUp item) {

        if (profile.getCurrency() >= item.getCost() &&
                profile.getPowerUpLevel(item.getPowerUpName()) < item.getMaxLevel()){
            int powerupLevel = profile.getPowerUpLevel(item.getPowerUpName());
            powerupLevel++;
            profile.subCurrency(item.getCost());
            profile.updatePowerUp(item.getPowerUpName(), powerupLevel);
            return true;
        }
        else{
            return false;
        }
    }
}

File: src/main/java/it/unibo/crabinv/model/powerups/ShopLogic.kt
package it.unibo.crabinv.model.powerups

import it.unibo.crabinv.model.core.save.UserProfile

/**
 * It's the logic of the shop.
 */
class ShopLogic : Shop {
    override fun purchase(
        profile: UserProfile,
        item: PowerUp,
    ): Boolean {
        if (profile.getCurrency() >= item.cost &&
            profile.getPowerUpLevel(item.powerUpType) < item.maxLevel
        ) {
            var powerupLevel = profile.getPowerUpLevel(item.powerUpType)
            powerupLevel++
            profile.subCurrency(item.cost)
            profile.updatePowerUp(item.powerUpType, powerupLevel)
            return true
        } else {
            return false
        }
    }
}
